language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Function.hs

```diff
@@ -247,3 +250,44 @@ getBuiltinFixN n = do
         TyAbs () s (Type ()) $
         LamAbs () h hTy $
         mkIterApp () (Var () h) branches
+
+data Function funName ann = Function
```

Comment:
Maybe put this and the two below in `MkPlc`? Seems useful in a similar way to `VarDecl` and co.

Reply:
Or maybe we need more stuff in a meta-PLC lib between the main one and the stdlib.